var/spack/repos/builtin/packages/synapsetool/package.py

```diff
@@ -58,26 +58,41 @@ class Synapsetool(CMakePackage):
 
     @property
     def libs(self):
-        """Export the synapse library (especially for neurodamus).
+        """Export the synapse library. Some packages like neurodamus
+        need explicit list of all libraries to link. Insted of searching
+        libraries in dependent package, synapsetool should provide all
+        libraries including dependencies.
         Sample usage: spec['synapsetool'].libs.ld_flags
         """
-        is_shared = '+shared' in self.spec
-        libs = find_libraries('libsyn2', root=self.prefix,
-                                  shared=is_shared, recursive=True)
-        if libs:
-            return libs
-        return None
+        spec = self.spec
+        is_shared = '+shared' in spec
+
+        # synapse tool libraries
+        libraries = find_libraries('libsyn2', root=self.prefix,
+                                  shared=is_shared, recursive=True).libraries
+
+        # boost libraries
```

Comment:
I don't think syntool static library `contains` dependencies.

Reply:
The lines spec['boost'] etc really must be conditional, otherwise the top-level package had to find himself all sub-dependencies. As noted above, RPATH is correctly set in shared libs, so we can rely on that to simplify dependency specification to only direct dependencies. I will make some more tests there